pkg/controller/master/image/vm_image_controller.go

```diff
@@ -111,7 +113,12 @@ func (h *vmImageHandler) initialize(image *harvesterv1.VirtualMachineImage) (*ha
 		h.imageController.EnqueueAfter(image.Namespace, image.Name, 10*time.Second)
 		return h.images.Update(image)
 	}
-	return h.createBackingImageAndStorageClass(image)
+
+	image, err = h.createBackingImageAndStorageClass(image)
+	if err != nil {
+		return h.images.Update(image)
+	}
+	return h.images.Update(image)
```

Comment:
It looks like no matter there is `err`, we do the same operation `h.images.Update(image)`. Do we want to call `toUpdate := handleFail(image.DeepCopy(), condition.Cond(harvesterv1.ImageInitialized), err)` here instead of in `createBackingImageAndStorageClass`.

Reply:
Actually the err nil check here is redundant, I've updated the code in https://github.com/harvester/harvester/pull/5498/commits/8adb4be68f1e06dd65b012db3e3dca773138ec80, many thanks